tests/src/threading_helpers.c

```diff
@@ -77,17 +77,38 @@ typedef struct {
 } mutex_functions_t;
 static mutex_functions_t mutex_functions;
 
-/** The total number of calls to mbedtls_mutex_init(), minus the total number
- * of calls to mbedtls_mutex_free().
+/**
+ *  The mutex used to guard live_mutexes below and access to the status variable
+ *  in every mbedtls_threading_mutex_t.
+ *  Note that we are not reporting any errors when locking and unlocking this
+ *  mutex. This is for a couple of reasons:
  *
- * Reset to 0 after each test case.
+ *  1. We have no real way of reporting any errors with this mutex - we cannot
+ *  report it back to the caller, as the failure was not that of the mutex
+ *  passed in. We could fail the test, but again this would indicate a problem
+ *  with the test code that did not exist.
+ *
+ *  2. Any failure to lock is unlikely to be intermittent, and will thus not
```

Comment:
Minor: Intermittent to me means that it is intermittent across full test runs and not within a single test run. Without Ryan's original comment I wouldn't have understood what we mean here. It is subjective, but maybe we could say something like:
> According to the PTHREAD specification if one of these operations fails then all future operations on the mutex will fail, and will thus ...